### DISCUSSION TITLE
gunicorn - IP vs domain name

### DISCUSSION THREAD

**Discussion by pmatuszy** [healthchecks/healthchecks#620] [Q&A]
Hi
a quick question - I can get only icons (static graphics) only when I use in my browser IP address of the web server.
When I used registered name of the web server no static graphics is server

physical server name: abc.domain.com   address 192.168.1.10
ping healthchecks.domain.com is resolved to 192.168.1.20     (it is an another ip address on that physical server)

I set env
```
export SITE_ROOT='http://healthchecks.domain.com:8000'
export PING_ENDPOINT=healthchecks.domain.com
```
then:
when I start
```
gunicorn hc.wsgi:application --threads 10 --bind healthchecks.domain.com:8000
```
I can't see any static graphics

but when i start 
```
gunicorn hc.wsgi:application --threads 10 --bind 192.168.1.20:8000
```
everything is looking nice.

Have you got any ideas why?

**Comment by cuu508** [comment]
I don't know what causes that difference.
Is there a particular reason why you want to use a hostname instead of ip address in the bind parameter?

**Comment by pmatuszy** [comment]
Thank you for your response. Answering your question:
there is no particular reason. I just changed it to the ip address and now healthckecks doesn't show any static graphics... 
Have you got any ideas why? So the website works but looks ugly just like on that screenshot below:

![xxxx](https://user-images.githubusercontent.com/58336955/158374803-6aef42cb-8feb-417a-b1f0-28c5ca34ecb1.jpg)

**Reply by pmatuszy** [reply]
I found the solution:
./manage.py migrate
./manage.py createsuperuser
python manage.py compress
**python manage.py collectstatic**

last line collectstatic solved the problem

**Comment by cuu508** [comment]
The first step to troubleshoot this would be to open browser's Developer tools, and in the network tab analyze the requests for static files. Does the URL look the way you would expect? What the server's response is?

**Reply by pmatuszy** [reply]
thank you for your reply - I really appreciate it. I did this - not sure though how it should look like. Will it help you to troubleshoot?

![image](https://user-images.githubusercontent.com/58336955/158389978-6cacc754-bd0b-4de4-9d71-f16a608c5c3f.png)
P.

**Reply by cuu508** [reply]
No, this does not help. Take a look at the requests for static files in the Network tab, and check for any errors or warnings in the Console below.

**Reply by pmatuszy** [reply]
ok - this is apparently the problem:
![image](https://user-images.githubusercontent.com/58336955/158574451-f3782558-cbde-47aa-af83-f86fd4f66d36.png)
Will it help with the investigation?
Paul

**Reply by cuu508** [reply]
Hi Paul,

I'm not sure why the icon font (icomoon.ttf, icomoon.woff) doesn't load with your configuration. I've tried to reproduce the problem a few different ways, but with no success. Clearly, we're doing something differently. 

Your next step would be to investigate the HTTP requests that load the icon font: check what domain name and what IP they are being loaded from. Verify that you can see these requests in gunicorn's access log. Unfortunately I don't have the time to help you with step-by-step instructions. You may want to look for someone with webdev experience to help you with this. If you do find the culprit and it is something that needs to be fixed in Healthchecks codebase or documentation, please file an issue (or, even better, make a PR!).
